libmcount/dynamic.c

```diff
@@ -420,7 +421,7 @@ static void patch_func_matched(struct mcount_dynamic_info *mdi,
 		    sym->type != ST_GLOBAL_FUNC)
 			continue;
 
-		if (!match_pattern_list(map, sym->name)) {
+		if (!match_pattern_list(map, sym->name, soname)) {
```

Comment:
Same comment as before.  Please move the code, you don't need to separate commits this way.